pkg/resource/resid.go

```diff
@@ -97,10 +97,25 @@ func (n ResId) Namespace() string {
 
 // CopyWithNewPrefix make a new copy from current ResId and append a new prefix
 func (n ResId) CopyWithNewPrefix(p string) ResId {
+	if p != "" && n.prefix != "" {
+		return ResId{gvk: n.gvk, name: n.name, prefix: p + ":" + n.prefix, namespace: n.namespace}
+	}
```

Comment:
readability:
```
func (n ResId) properPrefix(p string) string {
  if p == "" || n.prefix == "" {
    return p + n.prefix
  }
  return p + ":" + n.prefix
}
...
return ResId{gvk: n.gvk, name: n.name, prefix: n.fixPrefix(p), namespace: n.namespace}
```

Can properPrefix be used in FilterBy ?

Reply:
I name the function as `concatePrefix`.
Since this function manipulate a string given another string, we couldn't use this function directly in `FilterBy`. In `FilterBy`, we need a function to compare the leading part of two prefixes.